### PR TITLE
[GLUTEN-4946][CH] Fix avg(bigint) overflow

### PR DESCRIPTION
**Pull request by loudongfeng**
## What changes were proposed in this pull request?

cast avg input type to double when type is integer types: bigint/int/short...
Fixes: \#4946

## How was this patch tested?
 unit tests

### CODE REVIEW

**Comment by loudongfeng** [comment]
CI failures unrelated. Ready for review. @zzcclp thanks

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by binmahone — APPROVED** [review]

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 1]
```diff
@@ -1691,4 +1693,10 @@ object GlutenConfig {
       .doc("Force fallback for orc char type scan.")
       .booleanConf
       .createWithDefault(true)
+
+  val COLUMNAR_NATIVE_CAST_AGGREGATE_ENABLED =
+    buildConf("spark.gluten.sql.columnar.cast.avg")
+      .internal()
+      .booleanConf
+      .createWithDefault(true)
```
Comment (rui-mo) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:1701]: nit: do we need this config if the cast behavior is default on CH backend?

**Review by rui-mo — APPROVED** [review]

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala (hunk 1 above)
Comment (loudongfeng) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:1701]: We can disable this type cast when we are sure no overflow will happen, thereby improving the performance of some  cases.

Comment (zzcclp) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:1701]: @rui-mo agree with this? if yes, merge this pr ?

Comment (rui-mo) [reply shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:1701]: Yes, I understand. Thanks.

**Pull request merged**